src/routes_staff.py

```diff
@@ -1,253 +1,97 @@
-			for ext in extensions:
-				ext["_id"] = str(ext["_id"])
-			return jsonify(extensions), 200
-
-		@blueprint.route("/staff/course/<cid>/<aid>/extensions/", methods=["POST"])
-		@auth.require_auth
-		def staff_add_extension(netid, cid, aid):
-			if not verify_staff(netid, cid) or not verify_admin(netid, cid):
-				return abort(403)
-
-			def err(msg):
-				return msg, 400
-
-			assignment = db.get_assignment(cid, aid)
-			if not assignment:
-				return err("Invalid course or assignment. Please try again.")
-
-			if util.check_missing_fields(request.form, "netids", "max_runs", "start", "end"):
-				return err("Missing fields. Please try again.")
-
-			student_netids = request.form["netids"].replace(" ", "").split(",")
-			for student_netid in student_netids:
-				if not util.valid_id(student_netid) or not verify_student(student_netid, cid):
-					return err("Invalid or non-existent student NetID: %s" % student_netid)
-
-			try:
-				max_runs = int(request.form["max_runs"])
-				if max_runs < 1:
-					return err("Max Runs must be a positive integer.")
-			except ValueError:
-				return err("Max Runs must be a positive integer.")
-
-			start = util.parse_form_datetime(request.form["start"]).timestamp()
-			end = util.parse_form_datetime(request.form["end"]).timestamp()
-			if start >= end:
-				return err("Start must be before End.")
-
-			for student_netid in student_netids:
-				db.add_extension(cid, aid, student_netid, max_runs, start, end)
-			return "", 204
-
-		@blueprint.route("/staff/course/<cid>/<aid>/<run_id>/status/", methods=["GET"])
-		@auth.require_auth
-		def staff_get_run_status(netid, cid, aid, run_id):
-			if not verify_staff(netid, cid):
-				return abort(403)
-
-			status = bw_api.get_grading_run_status(cid, run_id)
-			if status:
-				return status, 200
-			return "", 400
-
-		@blueprint.route("/staff/course/<cid>/<aid>/<run_id>/log/", methods=["GET"])
-		@auth.require_auth
-		def staff_get_run_log(netid, cid, aid, run_id):
-			if not verify_staff(netid, cid):
-				return abort(403)
-
-			log = bw_api.get_grading_run_log(cid, run_id)
-			if log:
-				return jsonify(log), 200
-			return "", 400
-
-		@blueprint.route("/staff/course/<cid>/workers/", methods=["GET"])
-		@auth.require_auth
-		def staff_get_workers(netid, cid):
-			if not verify_staff(netid, cid):
-				return abort(403)
-
-			workers = bw_api.get_workers(cid)
-			if workers:
-				return jsonify(workers), 200
-			return "", 400
+    def __init__(self, blueprint):
+        @blueprint.route("/staff/", methods=["GET"])
+        @auth.require_auth
+        def staff_home(netid):
+            courses = db.get_courses_for_staff(netid)
+            version_code = 'unknown'
+            try:
+                git_hash = check_output('git rev-parse --short=8 HEAD'.split(' ')).decode('utf-8')
+                if len(git_hash) >= 8:
+                    version_code = git_hash[0:8]
+            except CalledProcessError:
+                pass
```

Comment:
How can we handle this error?

Reply:
We should probably log this error. Are the logs of this server currently displayed anywhere?

Reply:
@ayushr2 Do you know?

Reply:
I do not think the logs are displayed anywhere. They are just stored locally on a file I think.

Reply:
We can have the log output to a slack channel?

Reply:
the logs are very noisy. you can create an issue about dealing with logs better but that should not affect this PR. the error is handled afaik so resolving this.